### ISSUE TITLE
Feature request for StreamBuilder

### ISSUE THREAD

**Issue by Skquark** [martinrybak/futuristic#1]
This is the kind of addon I was looking for to solve an issue with my usage of Cloud Firestore, where my database streams keep getting rebuilt every time there's a state update, which has been causing glitches and wasted database calls.  Wanted an adaptation of this to work with StreamBuilder<QuerySnapshot> to do the same as it would for Futuristic<QuerySnapshot>..  The implementation of a FutureBuilder and StreamBuilder are almost the same, so I figured it'd be an easy change..

I was going to ask you to do it, but I peaked into your code and forked it over to add the feature myself easily.  Called it Streamistic, but could also be called Streamtastic or Streamist..  Your package name Futuristic sounds better, so feel free to include if you like. I basically replaced the word future with stream in your code and it pretty much worked out.  I haven't tested it yet, but looks like it should.  I could do a PR after I make sure, but my added file is here: https://github.com/Skquark/futuristic/blob/master/lib/streamistic.dart
Thanks for coming up with this method, clever implementation.  I have a lot of replacing of StreamBuilder to do in my projects (assuming it works).

**Comment by martinrybak** [comment]
Glad you found it useful! For the `StreamBuilder` equivalent, I can see the multiple widget callbacks being useful, but not necessarily the `_execute` part. You can't actually "start" a `Stream`; you can just start listening to it. What is your use case?

**Comment by Skquark** [comment]
Now that I've tried to use it, realizing it's not quite right yet and needs some tinkering.  My use case is for Cloud Firestore getting the snapshots from a Collection and displaying the snapshot data results. Here's a simple example of the normal method I'm wanting to replace:
```dart
           StreamBuilder<QuerySnapshot>(
                stream: Firestore.instance.collection("Events").snapshots(),
                builder: (BuildContext context, AsyncSnapshot<QuerySnapshot> snapshot) {
                  if (!snapshot.hasData || snapshot.connectionState == ConnectionState.waiting)
                    return CircularProgressIndicator();
                  if (snapshot.data.documents.length < 1) {
                    return Text( "No Events Added yet..." );
                  }
                  return Column(
                      children: snapshot.data.documents.map((resource) {
                    return EventTile(eventData: resource, context: context);
                  }).toList());
                });
```
First problem was with the type of streamBuilder being AsyncValueGetter<T> and giving it the Stream, gives this error "The return type 'Stream<QuerySnapshot>' isn't a 'Future<QuerySnapshot>', as required by the closure's context.".  I'm not sure what the equivalent of AsyncValueGetter is for the function of a Stream<T>, but would prefer that parameter to be a Stream type.
Also in the dataBuilder, I made the function type QuerySnapshot, but normally that would be AsyncSnapshot<QuerySnapshot> but I'm a little fuzzy on the ins-and-outs of it.  Maybe you understand how it comes together better.
You're right that it doesn't need the autoStart param, and I'm also thinking it'd be nice to have a method in there for emptyBuilder for when it hasData but the returned data.document is empty.
Needs a bit more re-tooling, but looks fixable.  Thanks, I'll play around with it a bit more, but you'd probably be able to get it functioning better than me..
